src/plugins/intel_gna/src/transformations/gather_sinking_unary.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/gather_sinking_unary.hpp"
+
+#include <transformations/utils/utils.hpp>
+#include <utility>
+#include <openvino/cc/ngraph/itt.hpp>
+
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/gather_sinking_utils.hpp"
+#include "transformations/rt_info/gather_sinking_attr.hpp"
+
+using namespace ov;
+using namespace ov::opset9;
+using namespace ov::pass::pattern;
+using namespace ov::op::util;
+using namespace gather_sinking;
+
+namespace {
+
+using NodePtr = std::shared_ptr<ov::Node>;
+using NodePair = std::pair<NodePtr, NodePtr>;
+
+/**
+ * @brief SwapNodes allows to perform swapping nodes even if there are more than one consumers but has less performance
+ *
+ * @param first_node first node pointer
+ * @param second_node first node pointer
+ * @return NodePair pair of nodes in new order that allows to register them in MatcherPass
+ */
+NodePair SwapNodes(NodePtr first_node, NodePtr second_node) {
+    auto second_node_inputs = second_node->input_values();
+    second_node_inputs[0] = first_node->input_value(0);
```

Comment:
If we process only one node input maybe it would be better do not touch others? 
e.g:
second_node->input_value[0] = first_node->input_value(0);

Reply:
You need to reconnect inputs/outputs of first_node and second_node. How can you do this without refer to their parent nodes? You cannot just write the code that you wrote. Please explain deeper your idea.